docs/operators/transform.md

```diff
@@ -0,0 +1,167 @@
+Transform - Operator Documentation
+==================================
+
+About Transform operator
+------------------------
+
+Transform means mapping of field expression from input to output or conversion of fields from one type to another.
+This operator is stateless. This operator receives objects on its input port; for each such input object, it creates a new output object whose fields are computed as expressions involving fields of the input object. 
+The types of the input and output objects are configurable as are the expressions used to compute the output fields. 
+
+The operator class is `TransformOperator` located in the package `com.datatorrent.lib.transform`.
+
+Use Case
+--------
+
+Consider the data received from input source and want to transform the data as below output schema while emitting to downstream operator. 
+
+Consider input objects with these fields:
+
+`{FirstName, LastName, Phone, DateOfBirth, Address}`
+ 
+ and output objects with fields: 
+ 
+ `{Name, Phone, Age, Address}`. 
+ 
+Suppose `Name` is a concatenation of `FirstName` and `LastName` and 
+        `Age` is computed by subtracting the `DateOfBirth` from the current year.
+
+These simple computations can be expressed as Java expressions where the input object is
+represented by $ and provided as configuration parameters as follows:
+
+```
+Name => {$.FirstName}.concat(\" \").concat({$.LastName})
+Age => (new java.util.Date()).getYear() - {$.dateOfBirth}.getYear()
+```
+
+#### Configuration Parameters
+-----------------------------
+
+-   ***expressionMap*** -   Map<String, String>
+    -   Mandatory Parameter
+    -   Specifies the map between the output field (key) and the expression used to compute it (value) using fields of the input Java object.
+
+-   ***expressionFunctions*** -   List<String>
+    -   List of imported classes or methods should be made available to expression to use. It overrides the default list.
+    -   Default Value = {java.lang.Math.*, org.apache.commons.lang3.StringUtils.*, org.apache.commons.lang3.StringEscapeUtils.*, org.apache.commons.lang3.time.DurationFormatUtils.*, org.apache.commons.lang3.time.DateFormatUtils.*}
+
+        Consider the input object with fields:
+        `{FirstName, LastName, StartDate}`. Here `StartDate` field is of type `org.joda.time.DateTime` 
+        and the output object with fields:
+        `{Name, isLeapYear}`. Here `isLeapYear` field is of type `boolean`.
+        Note that `org.joda.time.DateTime` class is not present in the default list. So, we need to add this library to `expressionFunctions` as below in populateDAG method:
+        
+```
```

Comment:
This section is coming out of nowhere and is not looking good in md. Please create a seperate section for code, properties and move it there..